icu4j/main/core/src/main/java/com/ibm/icu/text/DateFormatSymbols.java

```diff
@@ -2004,12 +2004,12 @@ protected void initializeData(ULocale desiredLocale, ICUResourceBundle b, String
         standaloneShortQuarters = arrays.get("quarters/stand-alone/abbreviated");
         standaloneNarrowQuarters = arrays.get("quarters/stand-alone/narrow");
 
-        abbreviatedDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/format/abbreviated"));
-        wideDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/format/wide"));
-        narrowDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/format/narrow"));
-        standaloneAbbreviatedDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/stand-alone/abbreviated"));
-        standaloneWideDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/stand-alone/wide"));
-        standaloneNarrowDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/stand-alone/narrow"));
+        abbreviatedDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/format/abbreviated"), null);
+        wideDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/format/wide"), abbreviatedDayPeriods);
+        narrowDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/format/narrow"), abbreviatedDayPeriods);
+        standaloneAbbreviatedDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/stand-alone/abbreviated"), abbreviatedDayPeriods);
+        standaloneWideDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/stand-alone/wide"), standaloneAbbreviatedDayPeriods);
+        standaloneNarrowDayPeriods = loadDayPeriodStrings(maps.get("dayPeriod/stand-alone/narrow"), standaloneAbbreviatedDayPeriods);
```

Comment:
Compare to the logic in `DateFormatSymbols::initializeData()`, ll. 2293-2311, in ICU4C.